tasks/src/main/scala/com/itv/bucky/tasks/Task.scala

```diff
@@ -0,0 +1,98 @@
+package com.itv.bucky.tasks
+
+import com.itv.bucky.PayloadMarshaller.StringPayloadMarshaller
+import com.itv.bucky.PublishCommandBuilder.publishCommandBuilder
+import com.itv.bucky._
+import com.itv.bucky.decl._
+import com.itv.bucky.pattern.requeue.{RequeuePolicy, _}
+import com.typesafe.scalalogging.StrictLogging
+
+import scala.concurrent.duration._
+import scala.language.higherKinds
+
+final case class TaskName(value: String)
```

Comment:
Nitpick: extend `AnyVal`?

Reply:
Sure. Would it be worth updating all existing case classes `RoutingKey, ExchangeName, QueueName, ConsumerTag, Delivery, etc` to also `extend AnyVal`?

Reply:
:+1: yeah think we should eventually, not necessarily as part of this PR